src/library_glfw.js

```diff
@@ -382,6 +385,38 @@ var LibraryGLFW = {
 #endif
     },
 
+    onGamepadConnected: function(event) {
+      if (!GLFW.active.joystickFunc) return;
+
+      // HTML5 Gamepad API reuses indexes
+      // glfw joystick API spreads them out
+      // http://www.glfw.org/docs/latest/input_guide.html#joystick
+      // "Once a joystick is detected, it keeps its assigned index until it is disconnected,
+      // so as joysticks are connected and disconnected, they will become spread out."
+      var joy = GLFW.active.joyLast++;
+      GLFW.active.joys[joy] = {
+        index: event.gamepad.index,
+        id: allocate(intArrayFromString(event.gamepad.id), 'i8', ALLOC_NORMAL),
+        buttons: allocate(new Array(event.gamepad.buttons.length), 'float', ALLOC_NORMAL),
+        axes: allocate(event.gamepad.axes, 'float', ALLOC_NORMAL)
+      };
+      GLFW.active.index2joy[event.gamepad.index] = joy;
+      Module['dynCall_vii'](GLFW.active.joystickFunc, joy, 0x00040001); // GLFW_CONNECTED
+    },
+
+    onGamepadDisconnected: function(event) {
+      if (!GLFW.active.joystickFunc) return;
+
+      var joy = GLFW.active.index2joy[event.gamepad.index];
+
+      Module['dynCall_vii'](GLFW.active.joystickFunc, joy, 0x00040002); // GLFW_DISCONNECTED
+
+      // TODO: free memory .axes, .buttons
```

Comment:
This can do `_free(GLFW.active.joys[joy].id);` and same for `buttons` and `axes`. (`ALLOC_NORMAL` is equivalent to a `_malloc()`)